mutablesecurity/solutions/implementations/duplicati/code.py

```diff
@@ -0,0 +1,693 @@
+class DefaultLogs(BaseLog):
+    class DefaultLogsFact(FactBase):
+        command = "cat /var/log/duplicati.log"
+
+        @staticmethod
+        def process(output: typing.List[str]) -> str:
+            return "\n".join(output)
+
+    IDENTIFIER = "logs"
+    DESCRIPTION = "Default log location"
+    FACT = DefaultLogsFact
+
+
+# Tests
+class SupportedArchitecture(BaseTest):
+    class SupportedArchitectureFact(BinaryArchitectureFact):
+        @staticmethod
+        def process(  # type: ignore[override]
+            output: typing.List[str],
+        ) -> bool:
+            architecture = BinaryArchitectureFact.process(output)
+
+            return architecture is not None
+
+    IDENTIFIER = "supported_architecture"
+    DESCRIPTION = "Checks if there is any build for this architecture."
+    TEST_TYPE = TestType.REQUIREMENT
+    FACT = SupportedArchitectureFact
+
+
+class ClientCommandPresence(BaseTest):
+    IDENTIFIER = "command"
+    DESCRIPTION = "Checks if the Duplici cli is registered as a command."
+    TEST_TYPE = TestType.PRESENCE
+    FACT = PresentCommand
+    FACT_ARGS = ("duplicati-cli --help",)
+
+
+class ClientEncryption(BaseTest):
+    class EncryptionTest(FactBase):
+        @staticmethod
+        def command() -> str:
+            backup_path = "/tmp/backup" + uuid.uuid4().hex
+            file_test = "/tmp/file.txt"
+            execute_command = [
+                "echo 'Hi' >> f{file_test}",
+                DuplicatiHelper._make_local_backup(file_test, backup_path)
+                + f" && dir {backup_path} |grep '*.f{EncryptionModule.get()}'",
+                f"rm -r {backup_path}",
+            ]
+
+            return execute_command
+
+        @staticmethod
+        def process(output: typing.List[str]) -> bool:
+            return int(output[0]) != 0
+
+    IDENTIFIER = "encryption_test"
+    DESCRIPTION = "Checks if the Duplici local encryption works"
+    TEST_TYPE = TestType.SECURITY
+    FACT = EncryptionTest
+
+
+# Solution class definition
+
+
+class Duplicati(BaseSolution):
+    INFORMATION = [
+        BinaryArchitecture,
+        CompressionModule,
+        EncryptionModule,
+        ExcludeFilesAttributes,
+        SkipFilesLarger,
+        RestorePath,
+        Passphrase,
+    ]
+    TESTS = [
+        ClientEncryption,
+        SupportedArchitecture,
+        ClientCommandPresence,
+    ]
+    LOGS = [
+        DefaultLogs,  # type: ignore[list-item]
+    ]
+    ACTIONS = [
+        LocalBackup,  # type: ignore[list-item]
+        RestoreLocalBackup,  # type: ignore[list-item]
+        SshBackup,  # type: ignore[list-item]
+        RestoreSshBackup,  # type: ignore[list-item]
+        GoogleDriveBackup,  # type: ignore[list-item]
+        RestoreGoogleDriveBackup,  # type: ignore[list-item]
+    ]
+
+    @staticmethod
+    @deploy
+    def _install() -> None:
+        architecture = BinaryArchitecture.get()
+        if not architecture:
+            raise IncompatibleArchitectureException()
```

Comment:
This will become redundant if you define the `REQUIREMENT` test I mentioned above.